lib/sass/file-to-class-names.ts

```diff
@@ -84,14 +116,15 @@ const classNameTransformer = (nameFormat: NameFormat): Transformer => {
     case "kebab":
     case "param":
       return (className) => paramCase(className);
-    case "camel":
-      return (className) => camelCase(className);
     case "snake":
       return (className) => snakeCase(className);
     case "dashes":
       return (className) =>
         /-/.test(className) ? camelCase(className) : className;
     case "none":
       return (className) => className;
+    case "camel":
+    default:
```

Comment:
Maybe would be better to make the `switch` exhaustive
```typescript

case "camel":
case "all":
    return (className) => camelCase(className);
default:
    const _: never = nameFormat;
```

Reply:
Hi @JohnGurin ,

Hmmm, in the current state of the code, the `all` never reaches to the classNameTransformer (it's [captured here](https://github.com/skovy/typed-scss-modules/pull/164/files/1a3fc54809b5a25023ff8ca4a4fee06fe303a87a#diff-15160ba27bbe478ce236bd2c20e0c022bac57ebf88b54aba5411cfca330f2d5eR53))

Also, is this just for the sake of TypeScript? Because `yargs` already does that check against the `NAME_FORMATS` [here](https://github.com/skovy/typed-scss-modules/blob/master/lib/cli.ts#L65).

Oh [and this](https://github.com/skovy/typed-scss-modules/pull/164/files/1a3fc54809b5a25023ff8ca4a4fee06fe303a87a#diff-15160ba27bbe478ce236bd2c20e0c022bac57ebf88b54aba5411cfca330f2d5eR51) already guarantees that the value(s) within the nameFormat is/are part of the NameFormat type 🤔

Reply:
>is this just for the sake of TypeScript

It is for when there is new variant, `ts` will report. `yargs` is for runtime check. Haven't dug into what `SASSOptions` does.
As `all` case stays separately, the type might be tweaked like this:
```typescript
const classNameTransformer = (nameFormat: Exclude<NameFormat, 'all'>): Transformer => {
```

Btw, I've just PRed with a fix to my prior PR, because, apparently, there is no tests for cli.

Reply:
And I don't see `consistent-return` eslint rule setup in the repo

Reply:
> It is for when there is new variant, `ts` will report. `yargs` is for runtime check. Haven't dug into what `SASSOptions` does. As `all` case stays separately, the type might be tweaked like this:
> 
> ```ts
> const classNameTransformer = (nameFormat: Exclude<NameFormat, 'all'>): Transformer => {
> ```
Sure ... I'll change...

> Btw, I've just PRed with a fix to my prior PR, because, apparently, there is no tests for cli.
What about these: https://github.com/skovy/typed-scss-modules/blob/master/__tests__/cli.test.ts ?

Reply:
Ricardo, I mean when I added `snake` case, I added it only to `NameFormat` type and missed `NAME_FORMATS` array which is used by `yargs`, CI tests were passed. Today I installed new version from npm and `snake` did not work, so I see it as tests does not cover `--nameFormat` cli option for `NameFormat`/`NAME_FORMATS` pair, and theirs types is not forcely synced by tools.

Reply:
Ah ... sorry, for a moment I perceived as "there are no tests for the CLI". 👍 All clear now.

Well, in order to implement the `never` situation and do a subset (via `Exclude<>`) I'll have to change the code a bit more...
If I have to do so, then I'll follow @skovy's request to transform the nameFormat argument into an `array: true` on the 
`cli.ts` (which will make it always an array) and go on from there... should simplify the code a bit.

Reply:
So I've refactored a bit the file. Instead of a switch we make use of a map of transformers (object literal) which still sucks as you have to maintain 2 places (`NAME_FORMATS` and `transformersMap`) but at least we get rid of the `switch`.

I also had to force the second parameter of the `main` call (the object literal) to be of the same type as expected on its definition (`Partial<CLIOptions>`).

Hope it's better now 👍

Reply:
Hi, Ricardo.
> ... as you still have to maintain 2 places ...

Not really, your type `Record<NameFormatsWithTransformer, Transformer>` does well - it prevents invalid variants and requires all of `NameFormatsWithTransformesrs` 👍

Reply:
[fed2e3b](https://github.com/skovy/typed-scss-modules/pull/164/commits/fed2e3b2e13f47bd5aa87fa93486c8f68ff61bd0)

Nice! Seems a lot better 👍  I also refactored a bit more to make use of `async .. await`